probe-rs/src/probe.rs

```diff
@@ -99,101 +99,110 @@ impl fmt::Display for BatchCommand {
-    InterfaceNotAvailable(&'static str),
-    /// Some interaction with the target registry failed.
-    /// This happens when an invalid chip name is given for example.
-    #[error("An error occurred while working with the registry")]
+    #[ignore_extra_doc_attributes]
+    InterfaceNotAvailable {
+        /// The name of the unsupported interface.
+        interface_name: &'static str,
+    },
+
+    /// An error occurred while working with the registry.
     Registry(#[from] RegistryError),
-    /// The debug probe does not support the speed that was chosen.
-    /// Try to alter the selected speed.
-    #[error("The requested speed setting ({0} kHz) is not supported by the probe")]
+
+    /// The probe does not support he requested speed setting ({0} kHz).
     UnsupportedSpeed(u32),
+
+    /// You need to be attached to the target to perform this action.
+    ///
     /// The debug probe did not yet perform the init sequence.
     /// Try calling [`DebugProbe::attach`] before trying again.
-    #[error("You need to be attached to the target to perform this action")]
+    #[ignore_extra_doc_attributes]
     NotAttached,
+
+    /// You need to be detached from the target to perform this action.
+    ///
     /// The debug probe already performed the init sequence.
     /// Try running the failing command before [`DebugProbe::attach`].
-    #[error("You need to be detached from the target to perform this action")]
+    #[ignore_extra_doc_attributes]
     Attached,
-    /// Performing the init sequence on the target failed.
-    /// Check the wiring before continuing.
-    #[error("Failed to find the target or attach to the target")]
+
+    /// Failed to find or attach to the target. Please check the wiring before retrying.
     TargetNotFound,
+
+    /// Error in previous batched command.
+    BatchError(BatchCommand),
+
+    /// The '{function_name}' functionality is not implemented yet.
+    ///
     /// The variant of the function you called is not yet implemented.
     /// This can happen if some debug probe has some unimplemented functionality for a specific protocol or architecture.
-    #[error("Some functionality was not implemented yet: {0}")]
-    NotImplemented(&'static str),
-    /// The called debug sequence is not supported on given probe.
-    /// This is most likely happening because you are using an ST-Link, which are severely limited in functionality.
-    /// If possible, try using another probe.
-    #[error("This debug sequence is not supported on the used probe: {0}")]
-    DebugSequenceNotSupported(&'static str),
-    /// An error occurred during the previously batched command.
-    #[error("Error in previous batched command")]
-    BatchError(BatchCommand),
-    /// The used functionality is not supported by the selected probe.
+    #[ignore_extra_doc_attributes]
+    NotImplemented {
+        /// The name of the unsupported functionality.
+        function_name: &'static str,
```

Comment:
I like that this is uniform now. I feel like that this error should provide a bit more info to the user than the function name (they cannot do anything with that). I wonder if we should make this a bit more descriptive or add an additional description field.

Reply:
This error feels a lot like a library-specific thing. If a user sees this, it's a bug IMO. Regardless, I feel like semantics belong to another PR.

Reply:
Ok, fair point! If this is just an internal thing it should be a panic imo. If it's exposed to library users the error should be more descriptive than an internal function name. But I guess it matches the name of the public function, so that's fine :)